bank_classes/account_class.rb

```diff
@@ -47,42 +48,65 @@ def self.find(id)
 # ### Error handling
 # - A new account cannot be created with initial negative balance - this will `raise` an `ArgumentError` (Google this)
   # begin
-    def open_deposit_check(initial_deposit)
-    transacting = true
-    while transacting do
-      if initial_deposit <= 0
-        raise Exception.new("New accounts must be opened with a positive balance.")
-      else
-        transacting = false
+    def open_deposit_check(initial_deposit, open_deposit_min)
+      if initial_deposit <= open_deposit_min
+        raise Exception.new("New accounts must be opened with a balance of at least #{open_deposit_min}.")
       end
       return initial_deposit.to_i
     end
-  end
 
 # - Should have a `withdraw` method that accepts a single parameter which represents the amount of money that will be withdrawn. This method should return the updated account balance.
-    def withdraw (withdraw_amount)
+    def withdraw (withdraw_amount, fee = 0, min_balance = 0)
       transacting = true
       while transacting do
-        withdraw_check = withdraw_amount
+
+        withdraw_check = (withdraw_amount + fee)
         if withdraw_check < 0
-          puts "You cannot withdaw #{withdraw_amount}. Your account currently have a blance of #{@balance}."
+          puts "You cannot withdaw a negative amount. Your account currently have a blance of #{@balance}."
           print "Would you like to withdaw a differnt amount? "
           diff_amount = gets.chomp.downcase
+
             if diff_amount == "yes" || diff_amount == "y"
-              puts "How much would you like to withdraw? "
-              withdraw_amount = gets.chomp.to_i
-              transacting = true
+              print "How much would you like to withdraw? >> "
+              withdraw_amount = gets.to_i
+            else
+              transacting = false
+            end
+
+        elsif (@balance - withdraw_check) <= min_balance
```

Comment:
Watch your indentation in this block, when you start an `if` block you only need to indent within the block rather than at the start of the block